Source/engine.cpp

```diff
@@ -548,26 +549,26 @@ void CelBlitOutlineTo(const CelOutputBuffer &out, BYTE col, int sx, int sy, cons
 	}
 }
 
-void SetPixel(const CelOutputBuffer &out, int sx, int sy, BYTE col)
+void SetPixel(const CelOutputBuffer &out, Point point, BYTE col)
 {
-	if (!out.in_bounds(sx, sy))
+	if (!out.in_bounds(point))
 		return;
 
-	*out.at(sx, sy) = col;
+	*out.at(point.x, point.y) = col;
 }
 
-void DrawLineTo(const CelOutputBuffer &out, int x0, int y0, int x1, int y1, BYTE color_index)
+void DrawLineTo(const CelOutputBuffer &out, Point a, Point b, BYTE color_index)
 {
-	int dx = x1 - x0;
-	int dy = y1 - y0;
+	int dx = b.x - a.x;
+	int dy = b.y - a.y;
 	int steps = abs(dx) > abs(dy) ? abs(dx) : abs(dy);
 	float ix = dx / (float)steps;
 	float iy = dy / (float)steps;
-	float sx = x0;
-	float sy = y0;
+	float sx = a.x;
+	float sy = a.y;
 
 	for (int i = 0; i <= steps; i++, sx += ix, sy += iy) {
-		SetPixel(out, sx, sy, color_index);
+		SetPixel(out, { (int)sx, (int)sy }, color_index);

```

Comment:
C-style cast

Reply:
Who can remember how to write the C++ style ones any way :)

Reply:
I usually go with `static_cast` first as it's the safest. Then `reinterpret_cast`/`const_cast`. I almost never use `dynamic_cast`. (I also often write `static_cats`)

Reply:
My older cat is pretty static at this point...